CMakeLists.txt

```diff
@@ -14,53 +14,46 @@ include(GNUInstallDirs)
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
 ##---------------------------------------------------------------------------##
-# Set up main options
+# Set up main options (inherit from Kokkos and Cabana CMake)
 ##---------------------------------------------------------------------------##
+find_package(Cabana REQUIRED)
+find_package(MPI REQUIRED)
+
 option(CabanaMD_ENABLE_Serial "Build CabanaMD with Serial support" ON)
 if( CabanaMD_ENABLE_Serial )
+  if( NOT Kokkos_ENABLE_SERIAL )
```

Comment:
You can do this now with Kokkos 3.0:
```
if( CabanaMD_ENABLE_${_device} )
     kokkos_check( DEVICES ${_device} )
endif()
```
where `${_device}` is whatever Kokkos device you have and it will check if it's been enabled. Note you could then do this all in a loop over possible device types you support.

Reply:
Added - this is much simpler